common/channel/src/libra_channel_test.rs

```diff
@@ -78,11 +78,11 @@ fn test_waker() {
         });
     });
     thread::sleep(Duration::from_millis(100));
-    sender.put(0, 0);
+    sender.put(0, 0).unwrap();
```

Comment:
you shouldn't need to use any `std::thread` things in here. you can have one future that reads 3 messages and another future that sends/waits 3 times. then join these two futures and await using a `tokio::Runtime` `block_on`

Reply:
another note: we should try do these kind of low-level waker/futures tests with many potential wake-interleavings using [`tokio-rs/loom`](https://github.com/tokio-rs/loom). I feel it will give us much greater assurance.

Reply:
Got rid of thread spawn/sleep and used tokio runtime and delay. I will try to add `loom` testing in another PR.